incubator/hnc/pkg/validators/hierarchy.go

```diff
@@ -127,6 +132,20 @@ func (v *Hierarchy) checkForest(hc *api.HierarchyConfiguration) ([]authzReq, adm
 	curParent := ns.Parent()
 	newParent := v.Forest.Get(hc.Spec.Parent)
 
+	// Check if requiredChild names in HierarchyConfiguration.Spec.RequiredChildren obey
```

Comment:
This check isn't actually checking the in-memory forest, so it shouldn't be in a function called `checkForest`. Maybe make a new function called `checkConfig`?

Reply:
Ok. I will change it according to your suggestion.